server/src/main/java/org/opensearch/ingest/IngestDocument.java

```diff
@@ -752,6 +753,7 @@ public Map<String, Object> getSourceAndMetadata() {
 
     @SuppressWarnings("unchecked")
     public static <K, V> Map<K, V> deepCopyMap(Map<K, V> source) {
+        CollectionUtils.ensureNoSelfReferences(source, "IngestDocument: Self reference present in object.");
```

Comment:
Can we add some unit tests here? I'm guessing without this check this code could result in a stack overflow?

Reply:
done